### PR TITLE
[win32] Fix core unit and integration tests

### PR DESCRIPTION
**Pull request by olivielpeau**
Allows the CI to pass (on AppVeyor) for the `default`, `core_integration` and `windows` test suites.

The tests that run only on unix are tagged with `@attr('unix')`. A few tests have been disabled on Windows for specific reasons that are documented in comments.

All the tests pass except for `test_parallelism` and `test_no_parallelism` of `tests.core.test_transaction` that are extremely flaky. They have been disabled on Windows for now.

cc @yannmh, @degemer

### CODE REVIEW

**Comment by yannmh** [comment]
Added a fix for the failing WMI tests.

File: tests/core/test_common.py [hunk 1]
```diff
@@ -240,8 +240,11 @@ def test_no_proxy(self):
         # Clear the env variables set
         del env["http_proxy"]
         del env["https_proxy"]
-        del env["HTTP_PROXY"]
-        del env["HTTPS_PROXY"]
+        if "HTTP_PROXY" in env:
+            # on some platforms (e.g. Windows) env var names are case-insensitive, so we have to avoid
```
Comment (degemer) [inline tests/core/test_common.py]: 😵

File: tests/core/test_flare.py [hunk 2]
```diff
@@ -168,13 +175,14 @@ def test_upload_with_case_proxy(self, mock_config, mock_tempdir, mock_stfrtime,
     def test_endpoint(self, mock_config, mock_temp, mock_stfrtime):
         f = Flare()
         f._ask_for_email = lambda: None
-        try:
+        f._open_tarfile()
+        f._tar.close()
+
+        with self.assertRaises(Exception) as cm:
```
Comment (degemer) [inline tests/core/test_flare.py]: 🍰

File: tests/core/test_flare.py [hunk 3]
```diff
@@ -233,12 +238,20 @@ def test_uri_no_verify_ssl(self, mock_config, mock_tempdir, mock_strftime):
         f.set_ssl_validation(request_options)
         self.assertFalse(request_options['verify'])
 
-    @attr(requires='core_integration')
```
Comment (degemer) [inline tests/core/test_flare.py]: Why did you remove all of them ?

File: transaction.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def compute_next_flush(self,max_delay):
         self._next_flush = newdate.replace(microsecond=0)
 
     def time_to_flush(self,now = datetime.utcnow()):
-        return self._next_flush < now
+        return self._next_flush <= now
```
Comment (degemer) [inline transaction.py]: Do you need this for a mock ? It's negligible anyway, just wondering.

File: utils/flare.py [hunk 5]
```diff
@@ -124,7 +125,12 @@ def collect(self):
         if not self._api_key:
             raise Exception('No api_key found')
         log.info("Collecting logs and configuration files:")
+        with self._open_tarfile():
```
Comment (degemer) [inline utils/flare.py]: 🍰

**Comment by degemer** [comment]
💯 
I think the Slack notifications from Appveyor are currently broken, could you take a pass at fixing them if you have some time ?

File: transaction.py (hunk 4 above)
Comment (degemer) [inline transaction.py]: Neverming, read your commit message.

File: tests/core/test_flare.py (hunk 3 above)
Comment (olivielpeau) [inline tests/core/test_flare.py]: I only left the `test_endpoint` test in `core_integration` because I thought that all the other ones were more unit tests or mock tests than actual integration tests. No other reason though.

**Comment by olivielpeau** [comment]
Thanks for the review @degemer! I've answered to your comments, amended the last commit message and re-enabled the Slack notifications :)

File: tests/core/test_flare.py (hunk 3 above)
Comment (degemer) [inline tests/core/test_flare.py]: You're right, I'm not sure why I put them in the first place.

**Comment by degemer** [comment]
Then :shipit:!

**Comment by olivielpeau** [comment]
CI is green, merging!

Merging 5 separate commits to keep a meaningful history.
